oeps/oep-0030-arch-pii-markup-and-auditing.rst

```diff
@@ -0,0 +1,268 @@
+
+- The LMS and Studio
+- IDAs such as Notes, Ecommerce, and Forums
+- Backend systems such as data analytics and operational logging
+- Third party services used to extend functionality of the services such as Optimizely, Google Analytics, and Sailthru
+
+Retirement
+----------
+Retirement is the Open edX process for "forgetting" a learner. It is triggered at the learner's request, and runs a sequence of automated steps to remove or anonymize learner PII across the Open edX ecosystem, in addition to alerting partners and 3rd party partners that they also need to forget any learner data they may have associated with the learner.
+
+The retirement process is designed to be flexible and extendable based on the needs of each Open edX installation. For instance, to retire new PII that is being added to an LMS database the developer may just need to add a method that listens to one of the existing LMS retirement Django signals in order to be added to the process. The retirement code and process are not part of this OEP and are documented here(TBD).
+
+Specification
+=============
+
+Developer Responsibility
+------------------------
+The responsibility for identifying and appropriately labeling PII rests on the developers working in Open edX code. When any new information is being saved to a persistent storage medium (ex. MySQL, Mongo, S3, reporting services, 3rd party marketing tools) the developer must identify whether any of that information may be (alone, or in conjunction with other stored data) PII, seeking legal assistance if necessary. Specifically the developer's responsibilities are:
+
+- Avoid storing PII when it is not necessary
+- Ensure that any PII that is stored will be retired upon learner request *before* that information is stored in a production environment
+- Ensure that any PII that is stored is annotated appropriately (see `Docstring Annotations`_ for details)
+- Run the PII documentation tool to update the PII documentation when they add, remove, or update a PII annotation (see `Documentation Tooling`_ for details)
+
+
+Code Reviewer Responsibility
+----------------------------
+It becomes the responsibility of code reviewers to confirm the developer assertions about the presence of PII in their pull request are accurate, that retirement steps and annotations are present and correct when necessary. This is especially important with pull requests coming from outside of edX, where the original developer may not know of this OEP and their responsibilities in regards to PII.
+
+Responsibility for Third-party Service Integrations
+---------------------------------------------------
+When dealing with third-parties that may store PII (ex. Optimizely, Google Analytics, Sailthru) the implementing developer(s) or team members should work with the drivers of the feature and legal to ensure that:
```

Comment:
I think "legal counsel" works better here to address maintainers of open edX instances that aren't edx.org - they may not have a legal department (which "legal" is presumably in reference to).